hms-app/src/pages/YourIdeas.tsx

```diff
@@ -21,75 +22,87 @@ function YourIdeas() {
     ideas: [],
   } as Hackathon)
 
-  const filteredIdeas = relevantIdeaList.filter((item) => {
-    return item.owner?.id.includes(userId)
+  const filteredIdeas = relevantIdeas.filter((item) => {
+    return item.owner?.user.id.includes(userID)
   })
 
-  const getHackathonID = (hackthonID: string) => {
+  const getParticipant = hackathons.participants?.filter((participant) => {
+    return participant.user.id.includes(userID)
+  })
```

Comment:
I'm pretty sure it returns an array, where later you do:
```
const getID = getParticipant?.map((participant) => {
    return participant.id
  })
```
This will return an collected array of ids.

To extract a singular participant matching a criteria, try:
```suggestion
  const getParticipant = hackathons.participants?.find((participant) => participant.user.id === userID)
```
[...]

Reply:
You can can see it easily if you add a type declaration `: ParticipantPreview` to left side.
```
const getParticipant : ParticipantPreview = hackathons.participants?.filter((participant) => {
    return participant.user.id.includes(userID)
  })
```
This tells you that:
`Type 'ParticipantPreview[] | undefined' is not assignable to type 'ParticipantPreview'`